### PR TITLE
Add mhlo::RngOp rewriter

### PR DESCRIPTION
**Pull request by tanyokwok**

### CODE REVIEW

File: tao_compiler/mlir/disc/BUILD [hunk 1]
```diff
@@ -420,6 +420,27 @@ cc_library(
     alwayslink = 1,
 )
 
+cc_library(
+    name = "rng_rewriter",
```
Comment (Yancey1989) [inline tao_compiler/mlir/disc/BUILD:424]: maybe we have other rewriter patterns from mhlo to disc-mhlo , so maybe we can rename to `disc_mhlo_rewirter`?

File: tao_compiler/mlir/disc/transforms/rng_rewriter.cc [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  explicit RngConvert(MLIRContext* context) : OpRewritePattern(context) {}
+
+  LogicalResult matchAndRewrite(mhlo::RngOp op,
+                                PatternRewriter& rewriter) const override {
+    if (op.rng_distribution() != ::mlir::mhlo::RngDistribution::UNIFORM) {
+      return failure();
+    }
+
+    auto cfg = PackRandomUniformBackendConfig(
+        rewriter.getIntegerAttr(rewriter.getI32Type(), 1),
```
Comment (Yancey1989) [inline tao_compiler/mlir/disc/transforms/rng_rewriter.cc:62]: Is that mean the random uniform has the same seed and generates the same value?
I prefer to lower disc-mhlo in torch-mlir conversion module, and we can use [GetNextSeedOp](https://github.com/llvm/torch-mlir/blob/main/include/torch-mlir/Dialect/TorchConversion/IR/TorchConversionOps.td#L219) to generate different random seed.

cc @wyzero any comment here?

Comment (wyzero) [reply tao_compiler/mlir/disc/transforms/rng_rewriter.cc:62]: +1

File: tao_compiler/mlir/disc/transforms/rng_rewriter.cc [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        cfg);
+    rewriter.replaceOp(op, {custom_call_op.getResult(0)});
+    return success();
+  }
+};
+
+struct RngRewriterPass : public RngRewriterPassBase<RngRewriterPass> {
+  void runOnOperation() override {
+    func::FuncOp func = getOperation();
+    // TODO: if needs to do const reformat, we need the xla_hlo.dot with its
```
Comment (wyzero) [inline tao_compiler/mlir/disc/transforms/rng_rewriter.cc:78]: remove unrelated comment.

File: tao_compiler/mlir/disc/transforms/rng_rewriter.cc (hunk 2 above)
Comment (fortianyou) [reply tao_compiler/mlir/disc/transforms/rng_rewriter.cc:62]: I think there is no need to set a random seed ahead of time. Instead, it's better to set a global seed at runtime.

File: tao_compiler/mlir/disc/transforms/rng_rewriter.cc (hunk 3 above)
Comment (fortianyou) [reply tao_compiler/mlir/disc/transforms/rng_rewriter.cc]: done

File: tao_compiler/mlir/disc/BUILD (hunk 1 above)
Comment (fortianyou) [reply tao_compiler/mlir/disc/BUILD]: done

**Pull request closed without merging**